http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java

```diff
@@ -134,6 +139,42 @@ public static FeignClientFactory of(
     }
 
     /**
+     * Constructs a dynamic proxy for the specified type using the supplied {@link ServiceConfiguration}. If the SSL
+     * configuration is not present in the {@link ServiceConfiguration}, the fallback SSL configuration will be used.
+     */
+    public <T> T createProxies(
+            ServiceConfiguration serviceConfig,
+            Optional<SslConfiguration> fallbackSslConfiguration,
+            Class<T> serviceClass) {
+
+        Optional<SSLSocketFactory> socketFactory = Optional.absent();
+        Optional<SslConfiguration> sslConfig = serviceConfig.security().or(fallbackSslConfiguration);
+
+        if (sslConfig.isPresent()) {
+            socketFactory = Optional.of(SslSocketFactories.createSslSocketFactory(sslConfig.get()));
+        }
+
+        Set<String> uris = Sets.newHashSet(serviceConfig.uris());
+        return FeignClients.standard().createProxy(socketFactory, uris, serviceClass);
+    }
+
+    /**
+     * Constructs a dynamic proxy for the specified type using the {@link ServiceConfiguration} obtained from the
+     * supplied {@link ServiceDiscoveryConfiguration} for the specified service name.
+     */
+    public <T> T createProxies(
```

Comment:
I had a similar opinion for the previous approach, it is really convenient, but we would have a  strong coupling between feign and service config.  Say we decide to use other client instead of feign,  it is harder to decouple them. Oh, take Retrofit client for example, we also need to be able to create retrofit client using the service config.  Also Service Config should not be changed often in the future.